scalr/resource_scalr_vcs_provider.go

```diff
@@ -0,0 +1,149 @@
+package scalr
+
+import (
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/scalr/go-scalr"
+	"log"
+)
+
+func resourceScalrVcsProvider() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalrVcsProviderCreate,
+		Read:   resourceScalrVcsProviderRead,
+		Update: resourceScalrVcsProviderUpdate,
+		Delete: resourceVcsProviderDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"vcs_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						string(scalr.Github),
+						string(scalr.GithubEnterprise),
+						string(scalr.Gitlab),
+						string(scalr.GitlabEnterprise),
+						string(scalr.BitbucketEnterprise),
+					},
+					false,
+				),
+			},
+			"token": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"account_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceScalrVcsProviderCreate(d *schema.ResourceData, meta interface{}) error {
+	scalrClient := meta.(*scalr.Client)
+	// Get attributes.
+	name := d.Get("name").(string)
+	token := d.Get("token").(string)
+	vcsType := scalr.VcsType(d.Get("vcs_type").(string))
+	options := scalr.VcsProviderCreateOptions{
+		Name:     &name,
+		VcsType:  vcsType,
+		Token:    token,
+		AuthType: "personal_token",
```

Comment:
@penja can't we use the constant introduced in client here?

Reply:
@brezitskiy yes we can